### PR TITLE
Revert "libvirt: 4.5.0 -> 4.6.0" and attempted fixes

### PR DESCRIPTION
**Pull request by xeji**
###### Motivation for this change

Libvirt 4.6.0 is broken (runtime errors). The cause seems to be a migration from yajl to jansson for JSON support. Upstream has reverted this after 4.6.0. Instead of applying a large number of patches, 
let's revert to libvirt 4.5.0 (which worked fine) and wait for the next upstream release.

This reverts the version bump and some unsuccessful attempts to fix it.

See also discussion  in #45409 . 

###### Things done

Built on NixOS.

---

cc @volth @LnL7 @fpletz @pikajude @VShell

### CODE REVIEW

**Comment by xeji** [comment]
@GrahamcOfBorg build libvirt

**Review by LnL7 — APPROVED** [review]

**Review by fpletz — APPROVED** [review]

File: pkgs/development/libraries/libvirt/default.nix [hunk 1]
```diff
@@ -60,15 +60,12 @@ in stdenv.mkDerivation rec {
...
-    substituteInPlace src/util/virjsoncompat.c --replace \
-        '"libjansson.so.4"' '"${jansson}/lib/libjansson${stdenv.targetPlatform.extensions.sharedLibrary}"'
-   '';
+  '';
 
   configureFlags = [
     "--localstatedir=/var"
     "--sysconfdir=/var/lib"
     "--with-libpcap"
-    "--with-qemu"
```
Comment (volth) [inline pkgs/development/libraries/libvirt/default.nix]: I'd keep ```--with-qemu```, 
although it is default in ```4.5.0``` but conditional in ```4.6.0``` and possible ```4.7.0``` and it is better to be sure we build ```libvirt``` with ```qemu``` than to let ```pkg-config``` decide

File: pkgs/development/libraries/libvirt/default.nix [hunk 2]
```diff
@@ -16,30 +16,30 @@ let
...
 
   nativeBuildInputs = [ makeWrapper pkgconfig ];
   buildInputs = [
     libxml2 gnutls perl python2 readline gettext libtasn1 libgcrypt yajl
-    libxslt xhtml1 perlPackages.XMLXPath curl libpcap jansson
+    libxslt xhtml1 perlPackages.XMLXPath curl libpcap
   ] ++ optionals (!buildFromTarball) [
     libtool autoconf automake
   ] ++ optionals stdenv.isLinux [
-    libpciaccess lvm2 utillinux systemd libnl numad zfs
```
Comment (volth) [inline pkgs/development/libraries/libvirt/default.nix]: One ```lvm2``` is enough.

File: pkgs/development/libraries/libvirt/default.nix (hunk 1 above)
Comment (volth) [reply pkgs/development/libraries/libvirt/default.nix]: I mean, on ```4.5.0 -> 4.7.0``` we likely will hit again this issue https://github.com/NixOS/nixpkgs/pull/44566#issuecomment-412222753

Comment (xeji) [reply pkgs/development/libraries/libvirt/default.nix]: Thanks, we may as well do that now :+1:

**Comment by volth** [comment]
```4.7.0-rc2``` is out, so ```4.7.0``` will be released in 1-2 days
Will we upgrade, or keep ```4.5.0``` for ```nixpkgs-18.09``` ?

**Comment by xeji** [comment]
Let's upgrade on master asap. We can always revert 18.09 to 4.5.0 in case of unexpected problems.

**Pull request merged**